Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Staging/SetPoints/SetpointController.mo

```diff
@@ -0,0 +1,596 @@
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.CapacityRequirement capReq1(
+    final avePer=avePer)
+    "Capacity requirement calculator"
+    annotation (Placement(transformation(extent={{-360,240},{-340,260}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.Capacities cap(
+    final nSta=nSta)
+    "Stage capacity calculator to to find design and minimum capacities for staging calculations"
+    annotation (Placement(transformation(extent={{-270,-180},{-250,-160}})));
+
+protected
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.Change cha(
+    final nSta=nSta,
+    final iniSta=iniSta,
+    final delStaCha=delStaCha)
+    "Stage change assignment"
+    annotation (Placement(transformation(extent={{-20,-180},{0,-160}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.BoilerIndices boiInd(
+    final nSta=nSta,
+    final nBoi=nBoi,
+    final staMat=staMat)
+    "Boiler index generator"
+    annotation (Placement(transformation(extent={{40,-210},{60,-190}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.Configurator conf(
+    final nSta=nSta,
+    final nBoi=nBoi,
+    final boiTyp=boiTyp,
+    final staMat=staMat,
+    final boiDesCap=boiDesCap,
+    final boiFirMin=boiFirMin)
+    "Configurator to decide stage availability based on boiler availability"
+    annotation (Placement(transformation(extent={{-360,-180},{-340,-160}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.Status sta(
+    final nSta=nSta,
+    final nBoi=nBoi,
+    final staMat=staMat)
+    "Status calculator to find next higher and lower available stages"
+    annotation (Placement(transformation(extent={{-310,-220},{-290,-200}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.Up staUp(
+    final nSta=nSta,
+    final fraNonConBoi=fraNonConBoi,
+    final fraConBoi=fraConBoi,
+    final sigDif=sigDif,
+    final delEffCon=delEffCon,
+    final TDif=TDif,
+    final TDifHys=TDifHys,
+    final delFaiCon=delFaiCon)
+    "Staging up calculator"
+    annotation (Placement(transformation(extent={{-140,-120},{-120,-88}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints.Subsequences.Down staDow(
+    final primaryOnly=primaryOnly,
+    final nSta=nSta,
+    final fraMinFir=fraMinFir,
+    final delMinFir=delMinFir,
+    final fraDesCap=fraDesCap,
+    final delDesCapNonConBoi=delDesCapNonConBoi,
+    final delDesCapConBoi=delDesCapConBoi,
+    final sigDif=sigDif,
+    final delBypVal=delBypVal,
+    final bypValClo=bypValClo,
+    final TCirDif=TCirDif,
+    final delTRetDif=delTRetDif,
+    final dTemp=dTemp,
+    final TDif=TDif,
+    final delFaiCon=delFaiCon,
+    bMinPriPumSpeSta=bMinPriPumSpeSta)
```

Comment:
looks like you forgot `final` here, also see if `b` needs to be `boi` - related to one of the previous comments.